core/src/Kokkos_View.hpp

```diff
@@ -101,6 +101,36 @@ void runtime_check_rank(const size_t rank, const size_t dyn_rank,
   }
 }
 
+/**
+ * @name Fencing can be required when allocating in @c CudaUVMSpace.
+ *
+ * If allocating in @c CudaUVMSpace, one must fence before and after
+ * the allocation to protect against possible concurrent access
+ * on the CPU and the GPU.
+ *
+ * By default, fencing is done on the default space instance.
+ * If a space instance is provided, it is used instead.
+ *
+ * @note Fence using the trait's execution space (which will be @c Kokkos::Cuda)
+ *       to avoid incomplete type errors from using @c Kokkos::Cuda directly.
+ */
+///@{
+#if defined(KOKKOS_ENABLE_CUDA)
+template <typename traits_t, typename alloc_prop_t>
+requires std::same_as<Kokkos::CudaUVMSpace,typename traits_t::device_type::memory_space>
+constexpr inline void allocation_fence_if_shared_space(alloc_prop_t&& prop, const char* msg)
```

Comment:
The `constexpr` is superfluous.  This is clearly not usable in constant expressions.

Reply:
Yes, I noticed. Removed :heavy_check_mark: